lib/chef/provisioning/aws_driver/super_lwrp.rb

```diff
@@ -7,30 +7,43 @@ class SuperLWRP < Chef::Resource::LWRPBase
     #
     # Add the :lazy_default and :coerce validation_opts to `attribute`
     #
-    def self.attribute(attr_name, validation_opts={})
-      lazy_default = validation_opts.delete(:lazy_default)
-      coerce = validation_opts.delete(:coerce)
-      if lazy_default || coerce
-        define_method(attr_name) do |arg=nil|
-          arg = instance_exec(arg, &coerce) if coerce && !arg.nil?
+    if self.respond_to?(:properties)
+      # in Chef 12.4+, properties replace attributes and these respond to
+      # coerce and default with a lazy block
+      def self.attribute(attr_name, validation_opts={})
+        lazy_default = validation_opts.delete(:lazy_default)
+        if lazy_default
+          Chef.log_deprecation("#{name} defines a lazy_default validation, instead specify default with a lazy block when Chef supports properties")
```

Comment:
Rightest thing to do here is probably to change the callers to use `default: lazy { ... }`. You can declare that with this in SuperLWRP:

```ruby
def self.lazy(&block)
  Chef::DelayedEvaluator.new(&block)
end
```

And you can check for it in the `else` case like this:

```ruby
if validation_opts[:default].is_a?(Chef::DelayedEvaluator)
  lazy_default = validation_opts.delete(:default)
end
```

Then just change the callers to use it.